plugins/urls/handlers/website.py

```diff
@@ -0,0 +1,266 @@
+
+        session = self.get_session(url, context)
+        session.get(url.text, headers=headers) \
+            .addCallback(self.callback, url, context, session) \
+            .addErrback(self.errback, url, context, session)
+
+        return False
+
+    def teardown(self):
+        # Save all our cookie stores
+        if self.global_session is not None:
+            self.global_session.cookies.save(ignore_discard=True)
+            self.global_session.close()
+
+        for session in self.group_sessions.itervalues():
+            session.cookies.save(ignore_discard=True)
+            session.close()
+
+    def reload(self):
+        self.teardown()
+        self.group_sessions = {}
+
+        self.global_session = Session()
+        self.global_session.cookies = self.get_cookie_jar("/global.txt")
+        self.global_session.session_type = "global"
+        self.global_session.cookies.set_mode(
+            # self.plugin.config["sessions"]["cookies"]["global"]
+            self.plugin.config.get("sessions", {})
+                .get("cookies", {})
+                .get("global", "discard")
+        )
+
+    def callback(self, response, url, context, session):
+        self.plugin.logger.trace(
+            "Headers: {0}", list(response.headers)
+        )
+        self.plugin.logger.trace("HTTP code: {0}", response.status_code)
+
+        if "content-type" not in response.headers:
+            response.headers["Content-Type"] = ""
+
+        content_type = response.headers["content-type"].lower()
+
+        if ";" in content_type:
+            content_type = content_type.split(";")[0]
+
+        if content_type not in context["config"]["content_types"]:
+            self.plugin.logger.debug(
+                "Unsupported Content-Type: %s"
+                % response.headers["content-type"]
+            )
+            return  # Not a supported content-type
+
+        soup = BeautifulSoup(response.text)
+
+        if soup.title and soup.title.string:
+            title = soup.title.string.strip()
+            title = re.sub("\s+", " ", title)
+            title = title
+
+            context["event"].target.respond(
+                to_unicode('"{0}" at {1}'.format(
+                    to_bytes(title), to_bytes(url.domain)
+                ))
+            )
+        else:
+            self.plugin.logger.debug("No title")
+
+        if session.session_type:
+            session.cookies.save(ignore_discard=True)
+
+            if len(session.cookies):
+                session.cookies.file_exists = True
+                session.cookies.load()
+
+    def errback(self, error, url, context, session):
+        self.plugin.logger.error("Error parsing URL")
+
+        if isinstance(error.value, ResponseNeverReceived):
+            for f in error.value.reasons:
+                f.printDetailedTraceback()
+                context["event"].target.respond(
+                    '"{0}" at {1}'.format(f.getErrorMessage(), url.domain)
```

Comment:
Is `url.domain` unicode or bytes? `to_bytes` is called on it above, which suggests it's unicode, but here it's being inserted into a bytestring. If it's unicode, change the format string to a unicode one and `to_unicode` the error message, since it's preferable to pass unicode strings to protocols (and in general).